File: Perl/tape-equilibrium.pl
#! /usr/bin/perl

use strict;
use warnings;

# http://codility.com/demo/take-sample-test/tapeequilibrium
sub sumarray{
    my @x = @_;
    print @x."aaa\n";
}

sub solution {
    my (@a)=@_;
    my $n = (scalar @a);
    my $head;
    my $tail;
    my $sum1;
    my $sum2;
    my $res;
    my $res2 = 2000;
    for (my $p = 1; $p < $n; $p++) {
        #print "\n".$p."  vvvvvvvvvvv\n";
        # How do I get part of an array in Perl ?????
        $sum1 = 0;
        map { $sum1 += $_ } @a[0 .. $p-1];
        #print $sum1."\n";

        $sum2 = 0;
        map {$sum2 += $_} @a[$p .. (scalar(@a)-1)];
        #print $sum2."\n";
        $res = abs($sum1 - $sum2);
        #print $res;
        $res2 = $res if ($res < $res2);
    }
    #print"\n";
    $res2;
}

print "ssssss\n";

print solution (3, 1, 2, 4, 3);


print "\nssssz\n";

print solution(-1000,1000);

File: Ruby/Codility/Lesson-1/tape-equilibrium.rb
def solution(a)
  n = a.size
  r = a.reverse
  p a
  for x in 1 .. n-1
    a[x] = a[x-1] + a[x]
    r[x] = r[x-1] + r[x]

  end
    puts "#{x} : #{a.inspect}  #{r.inspect}"
  r.reverse!
  #p a
  #p r
  r1 = 200
  for x in 0 .. a.size-2
    s=a[x]
    e=r[x+1]
    r2 = (s - e).abs
    #puts "#{s} #{e}  #{r2}"
    if r2 < r1
      r1 = r2
    end
  end
  r1
end

p solution [3, 1, 2, 4, 3]
p solution [-1000, 1000]
